### DISCUSSION TITLE
Share whole collection

### DISCUSSION THREAD

**Discussion by Code2Life** [toeverything/AFFiNE#6419] [Ideas]
Wanna use affine for cross-team knowledge share, build a systematical knowledge site is my primary goal, wondering if there is plan to make the whole collection sharable, so that we could replace things like **Notion Site**.

Some basic ideas of the user story:
1. Collection owner adds docs or rules to filter docs
2. Collection owner enables shared mode
3. the applications shows the branding / customization page, landing page editor etc.
4. Collection owner could use KB/blog templates or customize everything
5. Admin or Collection owner could setup public-collection site's policies like SEO options, IP/GEO allow list, RSS, custom-domain, comment-able, senmatic search engine integration etc.
6. Collection owner publish the whole collection, cloud backend will dynamically indexing new docs
7. Collection owner monitors the feedback and analytics data of the public site.

**Comment by EYHN** [comment]
Thanks for the suggestion, what you describe is what we call website building. 

Although we have many ideas, we don't have the manpower to implement them in the short term.
We will revisit the discussion when we start implementing it. 🙏

**Comment by gordielachance** [comment]
As a teacher, I would like to use AFFINE (online) for my course.

Being able to share pages is a huge step forward, but I think that's not really usable yet unless we can share collections and display a Table of Content.

Please work on this - and thanks for your awesome work !

**Comment by nigini** [comment]
I have just found out about AFFiNE, and it made my day. Amazing work y'all!

I am here to +1 the need for the "publish feature" improvement, which seems to still be restricted to one page at a time.
I have the exact same hopes as described by @Code2Life, but I think @gordielachance's simpler version (share collection/folder) sounds like a much simpler version and a super useful way to start!

Are there any news, pointers, or dev conversations around this in this community? I'd love to learn more about what it would take to make this happen.

**Comment by FreeSynergy** [comment]
It will be a killer feature, when you can use it as a public wiki.

But, if we do it smart, it can be much better. For Example:
We can use it like a wiki, like a document manager, like a canvas Board or all together … with a smart permission management, we can publish, we can create, we can make Workflows … 

As I said … this can be a killer feature … and boosts this app into a hogher level.

(Not, that this ist not good enough … but this could be better)

And Thanks for that very good word you do.

**Comment by antonvh** [comment]
Sounds like a great idea! I would love this. It would be nice to set a public collection as a home page for not-logged-in users. You could also write your own documentation using this. No more need to use RSPress.
